text/0605-eks-rewrite.md

```diff
@@ -0,0 +1,448 @@
+
+As a result, `AwsAuth` construct in the previous EKS module will not be provided in the new module.
+`grant()` functions are introduced to replace the awsAuth. It’s implemented using
+Access Entry.
+
+Grant Admin Access to an IAM role
+```
+cluster.grantAdmin('adminAccess', roleArn, eks.AccessScopeType.CLUSTER);
+```
+You can also use general `grantAccess()` to attach a policy to an IAM role/user.
+See https://docs.aws.amazon.com/eks/latest/userguide/access-policies.html for all access policies
+```
+# A general grant function is also provided
+cluster.grantAccess('adminAccess', roleArn, [
+  eks.AccessPolicy.fromAccessPolicyName('AmazonEKSClusterAdminPolicy', {
+    accessScopeType: eks.AccessScopeType.CLUSTER,
+  }),
+]);
+```
+
+### Use existing Cluster/Kubectl Handler
+
+This module allows defining Kubernetes resources such as Kubernetes manifests and Helm charts
+on clusters that are not defined as part of your CDK app.
+
+There are 2 scenarios here:
+1. Import the cluster without creating a new kubectl Handler
+```
+const cluster = eks.Cluster.fromClusterAttributes(this, 'MyCluster', {
+  clusterName: 'my-cluster-name',
+});
+```
+This imported cluster is not associated with a `Kubectl Handler`. It means we won't be able to 
+invoke `addManifest()` function on the cluster.
+
+To apply a manifest, you need to import the kubectl handler and attach it to the cluster
+```
+const kubectlProvider = eks.KubectlProvider.fromKubectlProviderArn(this, 'KubectlProvider', {
+  functionArn: ''
+});
+
+const cluster = eks.Cluster.fromClusterAttributes(this, 'MyCluster', {
+  clusterName: 'my-cluster-name',
+  kubectlProvider: kubectlProvider
+});
+
+cluster.addManifest();
+```
+
+2. Import the cluster and create a new kubectl Handler
+```
+const cluster = eks.Cluster.fromClusterWithKubectlProvider(this, 'MyCluster', {
+  clusterName: 'my-cluster-name',
+  kubectlProviderOptions: {
+    kubectlLayer: new KubectlV31Layer(this, 'kubectl'),
+  }
+});
+```
+This import function will always create a new kubectl handler for the cluster.
+
+#### Alternative Solution
+We can have one single `fromClusterAttributes()` and have different behaviors based on the input.
+
+- Import the cluster without kubectl Handler. It can't invoke AddManifest().
+```
+const cluster = eks.Cluster.fromClusterAttributes(this, 'MyCluster', {
+  clusterName: 'my-cluster-name',
+});
+```
+- Import the cluster and create a new kubectl Handler
+```
+const cluster = eks.Cluster.fromClusterAttributes(this, 'MyCluster', {
+  clusterName: 'my-cluster-name',
+  kubectlProviderOptions: {
+    kubectlLayer: new KubectlV31Layer(this, 'kubectl'),
+  }
+});
+```
+- Import the cluster/kubectl Handler
+```
+const kubectlProvider = eks.KubectlProvider.fromKubectlProviderArn(this, 'KubectlProvider', {
+  functionArn: ''
+});
+const cluster = eks.Cluster.fromClusterAttributes(this, 'MyCluster', {
+  clusterName: 'my-cluster-name',
+  kubectlProvider: kubectlProvider
+});
+```
+
+## Migration Guide
```

Comment:
I think we need answers to the following questions

Whats the feasibility of the migration? 
Whats possible whats breaking?
How much effort will we. need to upgrade a construct written in v1 and port it to v2.

Reply:
@dsilbergleithcu-godaddy Yes. As long as the cluster can be imported to CFN, I don't see a blocker of importing them into CDK and manage them with CDK.

Reply:
@QuantumNeuralCoder 
1. The feasibility of the migration depends on how complicated the stack is. The step that takes most time is to use `cdk import` to import the new resource.
2. The only step that could fail is `cdk import`. Users should use `cdk diff` to verify all diffs before moving forward.
3. These are manual steps but I don't think that will take lots of time. Should be able to migrate in 1 day or 2.